### PR TITLE
Improve DB connection error message

### LINKED ISSUE #159: Add better error message to `initDB` failures
Starting the server without the DB will result in a generic error that doesn't mention the DB:
`dial tcp 127.0.0.1:5432: getsockopt: connection refused`

### PR DESCRIPTION
**Pull request by carlosms**
Fix #159.

I also moved the DB initialization up. Because if we are going to abort on DB error, it does not make sense to establish connections with the analyzers before we know the DB is ok.

### CODE REVIEW

File: cmd/lookout/serve.go [hunk 1]
```diff
@@ -121,21 +136,6 @@ func (c *ServeCommand) Execute(args []string) error {
 		return err
 	}
 
-	db, err := c.initDB()
-	if err != nil {
```
[the remark this answers is not in the document]
Comment (bzz) [reply cmd/lookout/serve.go]: Does not seem like that, no? @carlosms

File: cmd/lookout/serve.go [hunk 2]
```diff
@@ -91,6 +91,21 @@ func (c *ServeCommand) Execute(args []string) error {
 		return err
 	}
 
+	db, err := c.initDB()
+	if err != nil {
+		return fmt.Errorf("Can't connect to the DB: %s", err)
```
Comment (bzz) [inline cmd/lookout/serve.go]: Could you check if that should follow https://github.com/golang/go/wiki/CodeReviewComments#error-strings ?

Comment (carlosms) [reply cmd/lookout/serve.go]: I usually follow that convention, but in this case I capitalized it to make it consistent with the previous messages, like "Can't open configuration file".
Do you think I should I change all 3 messages?

File: cmd/lookout/serve.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (carlosms) [reply cmd/lookout/serve.go]: ask @lookout-bot, not me!

what I'm more surprised about is that we are posting comments on _the left_ :thinking:

[the remark this answers is not in the document]
Comment (smacker) [reply cmd/lookout/serve.go]: #173

File: cmd/lookout/serve.go (hunk 2 above)
Comment (bzz) [reply cmd/lookout/serve.go]: IDK, up to you - but consistency wins and this also might be top-level error

Comment (carlosms) [reply cmd/lookout/serve.go]: merged, we can change this later

**Pull request merged**